haystack/dataclasses/breakpoints.py

```diff
@@ -202,34 +193,67 @@ class PipelineSnapshot:
 
     :param pipeline_state: The state of the pipeline at the time of the snapshot.
     :param break_point: The breakpoint that triggered the snapshot.
+    :param agent_snapshot: Optional agent snapshot if the breakpoint is an agent breakpoint.
     :param timestamp: A timestamp indicating when the snapshot was taken.
+    :param original_input_data: The original input data provided to the pipeline.
+    :param ordered_component_names: A list of component names in the order they were visited.
+    :param include_outputs_from: Set of component names whose outputs should be included in the pipeline results.
+    :param intermediate_outputs: Dictionary containing outputs from components that are in the include_outputs_from set.
+    :param pipeline_outputs: Dictionary containing the final outputs of the pipeline up to the breakpoint.
     """
 
     pipeline_state: PipelineState
     break_point: Union[AgentBreakpoint, Breakpoint]
     agent_snapshot: Optional[AgentSnapshot] = None
     timestamp: Optional[datetime] = None
+    original_input_data: Dict[str, Any] = field(default_factory=dict)
+    ordered_component_names: List[str] = field(default_factory=list)
```

Comment:
I think I'd make these required inputs and put them above `pipeline_state` since they should always be present when running a pipeline.

Reply:
But that would be the reason for being in the snapshot, as I'm seeing it:


- `PipelineSnapShot`: all the parameters that are passed at init + run time and values generated once that don't change.
- `PipelineState`: holds all the data generated during execution and captured by a breakpoint.

Do you agree?

Reply:
Right sorry I do agree that we should keep in them in PipelineSnapshot. I more meant that we should make them required inputs to PipelineSnapshot and not have default values.

Reply:
ah got it - agree 👍🏽